impl/src/main/java/org/jboss/cdi/tck/tests/lookup/dynamic/handle/InstanceHandleTest.java

```diff
@@ -0,0 +1,139 @@
+        Client client = getContextualReference(Client.class);
+        ActionSequence.reset();
+        assertNotNull(client);
+        assertTrue(client.getAlphaInstance().isResolvable());
+        assertFalse(client.getBigDecimalInstance().isResolvable());
+    }
+
+    //@SpecAssertion(section = PROGRAMMATIC_LOOKUP, id = "TODO")
+    @Test
+    public void testGetHandler() {
+        Client client = getContextualReference(Client.class);
+        BeanManager beanManager = getCurrentManager();
+
+        ActionSequence.reset();
+        assertNotNull(client);
+
+        Bean<?> alphaBean = beanManager.resolve(beanManager.getBeans(Alpha.class));
+        Instance<Alpha> instance = client.getAlphaInstance();
+
+        Instance.Handle<Alpha> alpha1 = instance.getHandle();
+        assertEquals(alphaBean, alpha1.getBean());
+        assertEquals(Dependent.class, alpha1.getBean().getScope());
+
+        String alpha2Id;
+
+        // Test try-with-resource
+        try (Instance.Handle<Alpha> alpha2 = instance.getHandle()) {
+            alpha2Id = alpha2.get().getId();
+            assertFalse(alpha1.get().getId().equals(alpha2Id));
+        }
+
+        List<String> sequence = ActionSequence.getSequenceData();
+        assertEquals(1, sequence.size());
+        assertEquals(alpha2Id, sequence.get(0));
+
+        alpha1.destroy();
+        // Subsequent invocations are no-op
+        alpha1.destroy();
+
+        sequence = ActionSequence.getSequenceData();
+        assertEquals(2, sequence.size());
+
+        // Test normal scoped bean is also destroyed
+        Instance<Bravo> bravoInstance = client.getInstance().select(Bravo.class);
+        String bravoId = bravoInstance.get().getId();
+        try (Instance.Handle<Bravo> bravo = bravoInstance.getHandle()) {
+            assertEquals(bravoId, bravo.get().getId());
+            ActionSequence.reset();
+        }
+        sequence = ActionSequence.getSequenceData();
+        assertEquals(1, sequence.size());
+        assertEquals(bravoId, sequence.get(0));
+    }
+
+    //@SpecAssertion(section = PROGRAMMATIC_LOOKUP, id = "TODO")
+    @Test
+    public void testGetAfterDestroyingContextualInstance() {
+        ActionSequence.reset();
+        Client client = getContextualReference(Client.class);
+        assertNotNull(client);
+
+        Instance.Handle<Alpha> alphaHandle = client.getAlphaInstance().getHandle();
+        // trigger bean creation
+        alphaHandle.get();
+        // trigger bean destruction
+        alphaHandle.destroy();
+        // verify that the destruction happened
+        List<String> sequence = ActionSequence.getSequenceData();
+        assertEquals(1, sequence.size());
+
+        // try to invoke Handle.get() again; this should throw an exception
+        try {
+            alphaHandle.get();
+            fail("Invoking Handle.get() after destroying contextual instance should throw an exception.");
+        } catch (IllegalStateException e) {
```

Comment:
I assume you're going to submit a separate PR to CDI for this?

Reply:
https://github.com/eclipse-ee4j/cdi/pull/555